misaligned.cpp

```diff
@@ -1,21 +1,70 @@
 #include <assert.h>
 #include <iostream>
+#include <sstream>
+#include <string>
+#include <algorithm>
+#include <tuple>
 
-int printColorMap() {
+int printColorMap(std::ostream& out) {
     const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
     const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
     int i = 0, j = 0;
     for (i = 0; i < 5; i++) {
         for (j = 0; j < 5; j++) {
-            std::cout << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
+            out << i * 5 + j << " | " << majorColor[i] << " | " << minorColor[i] << "\n";
         }
     }
     return i * j;
 }
 
+// Test - core part
+std::tuple<int, int, int, int> testPrintColorMap() {
+    std::stringstream ss;
+    printColorMap(ss);
+    // test
+    const char* majorColor[] = { "White", "Red", "Black", "Yellow", "Violet" };
+    const char* minorColor[] = { "Blue", "Orange", "Green", "Brown", "Slate" };
+    int nCntNumMatch = 0, nCntMajorMatch = 0, nCntMinorMatch = 0, nCntFullMatch = 0;
+    std::string inputStr;
+    int i = 0, j = 0;
+    while (std::getline(ss, inputStr, '\n')) {
+        std::string numStr,  majorStr, minorStr, partitionStr;
+        std::stringstream inputSS(inputStr);
+        inputSS >> numStr >> partitionStr >> majorStr >> partitionStr >> minorStr;
+        auto expectedNumStr = std::to_string(i * 5 + j);

```

Comment:
The code has a bug, but dont compromise the test integrity. The expected number is 1 to 25, while this will expect 0 to 24, which is wrong

Reply:
Rectified.